flink-core/src/main/java/org/apache/flink/api/common/operators/ResourceSpec.java

```diff
@@ -61,18 +79,17 @@
 	/** How many state size in mb are used */
 	private final int stateSizeInMB;
 
+	private final Map<String, Resource> extendedResources = new HashMap<>(1);
```

Comment:
This violates the serializability of `ResourceSpec` if `Resource` itself is not serializable.

Reply:
done, and add a test for it.